docs/content/concepts/assets/software-defined-assets.mdx

```diff
@@ -260,6 +260,34 @@ def downstream(may_not_materialize):
     return may_not_materialize + [5]
 ```
 
+### Asset code versions
+
+Assets may be assigned a `code_version`:
+
+```python file=/concepts/assets/code_versions.py startafter=start_single_asset endbefore=end_single_asset
+@asset(code_version="1")
+def asset_with_version():
+  return 100
+```
+
+When an asset with a code version is materialized, the generated `AssetMaterialization` is tagged with the version. The code version together with the <PyObject object="LogicalVersion" /> values of upstream assets are hashed to create a `LogicalVersion` for the asset, also stored on the `AssetMaterialization`, which is used to track asset staleness. Updating the `code_version` will change the projected `LogicalVersion` of the asset for a new materialization, thus rendering it stale.
```

Comment:
I think there's a way to explain this that doesn't require readers to understand the details of the implementation / hashing etc.  E.g. "An asset that has a different code version than the code version used for its most recent materialization will be considered _stale_. Any assets downstream of a stale asset are also considered stale.".